--- client/command.c	2019-12-13 16:16:17.000000000 +0700
+++ client/command1.c	2021-12-01 17:48:36.081752975 +0700
@@ -29,7 +29,7 @@
 #include <string.h>
 #include <getopt.h>
 #include <ell/ell.h>
-#include <readline/readline.h>
+#include <editline/readline.h>
 
 #include "client/command.h"
 #include "client/display.h"

--- client/display.c	2019-12-13 16:16:17.000000000 +0700
+++ client/display1.c	2021-12-02 08:04:38.210562934 +0700


@@ -572,7 +572,9 @@
 		rl_replace_line("", 0);
 		rl_redisplay();
 
+#ifdef rl_erase_empty_line
 		rl_erase_empty_line = 0;
+#endif
 		rl_set_prompt(prompt);
 	} else {
 		rl_callback_handler_install(prompt, readline_callback);
@@ -609,9 +611,9 @@
 		l_queue_push_tail(display_refresh.redo_entries, prompt);
 		display_refresh.undo_lines++;
 	}
-
+#ifdef rl_erase_empty_line
 	rl_erase_empty_line = 1;
-
+#endif
 	rl_replace_line(agent_saved_input->line, 0);
 	rl_point = agent_saved_input->point;
 
@@ -677,7 +679,9 @@
 	rl_completion_display_matches_hook = display_completion_matches;
 
 	rl_completer_quote_characters = "\"";
+#ifdef rl_erase_empty_line
 	rl_erase_empty_line = 1;
+#endif
 	rl_callback_handler_install("Waiting for IWD to appear...",
 							readline_callback);
 
--- client/display.c	2021-12-02 08:06:04.723191597 +0700
+++ client/display1.c	2021-12-02 08:11:05.354579244 +0700
@@ -29,8 +29,8 @@
 #include <signal.h>
 #include <sys/stat.h>
 
-#include <readline/history.h>
-#include <readline/readline.h>
+#include <histedit.h>
+#include <editline/readline.h>
 #include <ell/ell.h>
 
 #include "client/agent.h"
--- client/display.c	2021-12-02 08:12:25.664015612 +0700
+++ client/display1.c	2021-12-02 08:18:04.675653596 +0700
@@ -67,8 +67,10 @@
 {
 	struct saved_input *input;
 
+#ifdef RL_STATE_DONE
 	if (RL_ISSTATE(RL_STATE_DONE))
 		return NULL;
+#endif
 
 	input = l_new(struct saved_input, 1);
 
--- Makefile1.am	2021-12-02 08:34:41.009262444 +0700
+++ Makefile.am	2020-03-25 16:37:15.000000000 +0700
@@ -279,7 +279,7 @@
 			client/network.h client/network.c \
 			client/properties.h client/properties.c \
 			client/wsc.c client/station.c
-client_iwctl_LDADD = $(ell_ldadd) -lreadline
+client_iwctl_LDADD = $(ell_ldadd) -ledit
 
 if MANUAL_PAGES
 man_MANS += client/iwctl.1
@@ -502,7 +502,7 @@
 				client/display.h client/display.c \
 				client/network.h client/network.c \
 				client/properties.h client/properties.c
-unit_test_client_LDADD = $(ell_ldadd) -lreadline
+unit_test_client_LDADD = $(ell_ldadd) -ledit
 endif
 
 unit_test_sae_SOURCES = unit/test-sae.c \
--- client/display.c	2021-12-02 08:40:42.198927413 +0700
+++ client/display1.c	2021-12-02 08:53:47.320113658 +0700
@@ -75,9 +75,17 @@
 	input = l_new(struct saved_input, 1);
 
 	input->point = rl_point;
+#ifdef rl_copy_text
 	input->line = rl_copy_text(0, rl_end);
+#endif
+
+#ifdef rl_save_prompt
 	rl_save_prompt();
+#endif
+
+#ifdef rl_replace_line
 	rl_replace_line("", 0);
+#endif
 	rl_redisplay();
 
 	return input;
@@ -87,9 +95,13 @@
 {
 	if (!input)
 		return;
-
+#ifdef rl_restore_prompt
 	rl_restore_prompt();
+#endif
+
+#ifdef rl_replace_line
 	rl_replace_line(input->line, 0);
+#endif
 	rl_point = input->point;
 	rl_forced_update_display();
 
@@ -367,7 +379,9 @@
 	char line[LINE_LEN];
 	size_t index;
 	size_t line_used;
+#ifdef rl_copy_text
 	char *input = rl_copy_text(0, rl_end);
+#endif
 
 	prompt = l_strdup_printf("%s%s\n", IWD_PROMPT, input);
 	l_free(input);
@@ -569,9 +583,13 @@
 		agent_saved_input = l_new(struct saved_input, 1);
 
 		agent_saved_input->point = rl_point;
+#ifdef rl_copy_text
 		agent_saved_input->line = rl_copy_text(0, rl_end);
+#endif
 		rl_set_prompt("");
+#ifdef rl_replace_line
 		rl_replace_line("", 0);
+#endif
 		rl_redisplay();
 
 #ifdef rl_erase_empty_line
@@ -605,7 +623,9 @@
 		return;
 
 	if (display_refresh.cmd) {
+#ifdef rl_copy_text
 		char *text = rl_copy_text(0, rl_end);
+#endif		
 		char *prompt = l_strdup_printf(COLOR_BLUE "%s " COLOR_OFF
 							"%s\n", label, text);
 		l_free(text);
@@ -616,7 +636,9 @@
 #ifdef rl_erase_empty_line
 	rl_erase_empty_line = 1;
 #endif
+#ifdef rl_replace_line
 	rl_replace_line(agent_saved_input->line, 0);
+#endif
 	rl_point = agent_saved_input->point;
 
 	l_free(agent_saved_input->line);
--- client/display.c	2021-12-02 08:55:48.056444978 +0700
+++ client/display1.c	2021-12-02 09:07:43.502500604 +0700
@@ -381,6 +381,8 @@
 	size_t line_used;
 #ifdef rl_copy_text
 	char *input = rl_copy_text(0, rl_end);
+#else
+	char *input;
 #endif
 
 	prompt = l_strdup_printf("%s%s\n", IWD_PROMPT, input);
@@ -621,7 +623,7 @@
 
 	if (!agent_saved_input)
 		return;
-
+#if 0
 	if (display_refresh.cmd) {
 #ifdef rl_copy_text
 		char *text = rl_copy_text(0, rl_end);
@@ -633,21 +635,31 @@
 		l_queue_push_tail(display_refresh.redo_entries, prompt);
 		display_refresh.undo_lines++;
 	}
+#endif
+
 #ifdef rl_erase_empty_line
 	rl_erase_empty_line = 1;
 #endif
 #ifdef rl_replace_line
 	rl_replace_line(agent_saved_input->line, 0);
-#endif
 	rl_point = agent_saved_input->point;
+#endif
 
 	l_free(agent_saved_input->line);
 	l_free(agent_saved_input);
-	agent_saved_input = NULL;
+	agent_saved_input = NULL;
 
 	rl_set_prompt(IWD_PROMPT);
 }
 
+#ifndef rl_crlf
+static int rl_crlf(void) {
+    if (rl_outstream != NULL)
+        putc('\n', rl_outstream);
+    return 0;
+}
+#endif
+
 void display_quit(void)
 {
 	rl_crlf();
@@ -700,7 +712,9 @@
 	resize_signal = l_signal_create(SIGWINCH, signal_handler, NULL, NULL);
 
 	rl_attempted_completion_function = command_completion;
+#ifdef rl_completion_display_matches_hook
 	rl_completion_display_matches_hook = display_completion_matches;
+#endif
 
 	rl_completer_quote_characters = "\"";
 #ifdef rl_erase_empty_line
--- client/display.c	2021-12-02 09:33:33.836903190 +0700
+++ client/display1.c	2021-12-02 09:35:37.407300087 +0700
@@ -652,17 +652,11 @@
 	rl_set_prompt(IWD_PROMPT);
 }
 
-#ifndef rl_crlf
-static int rl_crlf(void) {
-    if (rl_outstream != NULL)
-        putc('\n', rl_outstream);
-    return 0;
-}
-#endif
-
 void display_quit(void)
 {
+#ifdef rl_crlf
 	rl_crlf();
+#endif
 }
 
 static void signal_handler(void *user_data)
--- client/command.c	2021-12-02 09:38:27.027818803 +0700
+++ client/command1.c	2021-12-02 09:43:36.658955601 +0700
@@ -323,7 +323,9 @@
 		goto done;
 	}
 
+#ifdef rl_copy_text
 	prompt = rl_copy_text(0, rl_end);
+#endif
 
 	family = strtok_r(prompt, " ", &args);
 	if (!family)
