Fix CODESET, UT_HOSTSIZE, UT_LINESIZE & UT_NAMESIZE macros
--- proc/escape.c	2022-08-20 11:59:27.398454438 +0700
+++ proc/escape.c1	2022-08-20 11:59:04.808604958 +0700
@@ -21,6 +21,7 @@
 #include <sys/types.h>
 #include <string.h>
 #include <limits.h>
+#include <langinfo.h>
 #include "procps.h"
 #include "escape.h"
 #include "readproc.h"
@@ -30,7 +31,6 @@
 # include <wctype.h>
 # include <stdlib.h>  /* MB_CUR_MAX */
 # include <ctype.h>
-# include <langinfo.h>
 #endif
 
 #define SECURE_ESCAPE_ARGS(dst, bytes, cells) do { \
--- w.c	2021-02-09 17:11:25.000000000 +0700
+++ w.c1	2022-08-20 12:10:39.889906758 +0700
@@ -55,11 +55,7 @@
 #include <termios.h>
 #include <time.h>
 #include <unistd.h>
-#ifdef HAVE_UTMPX_H
-#	include <utmpx.h>
-#else
-#	include <utmp.h>
-#endif
+#include <utmp.h>
 #include <arpa/inet.h>
 
 static int ignoreuser = 0;	/* for '-u' */
