--- Makefile	2022-01-31 22:54:59.004285221 +0700
+++ Makefile1	2022-01-31 22:54:50.018038716 +0700
@@ -442,29 +442,30 @@
 
 # Make variables (CC, etc...)
 CPP		= $(CC) -E
+# Pick up system variables
 ifneq ($(LLVM),)
-CC		= clang
-LD		= ld.lld
-AR		= llvm-ar
-NM		= llvm-nm
-OBJCOPY		= llvm-objcopy
-OBJDUMP		= llvm-objdump
-READELF		= llvm-readelf
-STRIP		= llvm-strip
+CC		?= clang
+LD		?= ld.lld
+AR		?= llvm-ar
+NM		?= llvm-nm
+OBJCOPY		?= llvm-objcopy
+OBJDUMP		?= llvm-objdump
+READELF		?= llvm-readelf
+STRIP		?= llvm-strip
 else
-CC		= $(CROSS_COMPILE)gcc
-LD		= $(CROSS_COMPILE)ld
-AR		= $(CROSS_COMPILE)ar
-NM		= $(CROSS_COMPILE)nm
-OBJCOPY		= $(CROSS_COMPILE)objcopy
-OBJDUMP		= $(CROSS_COMPILE)objdump
-READELF		= $(CROSS_COMPILE)readelf
-STRIP		= $(CROSS_COMPILE)strip
+CC		?= $(CROSS_COMPILE)gcc
+LD		?= $(CROSS_COMPILE)ld
+AR		?= $(CROSS_COMPILE)ar
+NM		?= $(CROSS_COMPILE)nm
+OBJCOPY		?= $(CROSS_COMPILE)objcopy
+OBJDUMP		?= $(CROSS_COMPILE)objdump
+READELF		?= $(CROSS_COMPILE)readelf
+STRIP		?= $(CROSS_COMPILE)strip
 endif
 PAHOLE		= pahole
 RESOLVE_BTFIDS	= $(objtree)/tools/bpf/resolve_btfids/resolve_btfids
 LEX		= flex
-YACC		= bison
+YACC		?= bison
 AWK		= awk
 INSTALLKERNEL  := installkernel
 DEPMOD		= depmod
